Add filtering/selection by peakmax and brightest

mcara opened the pull request:
This PR adds two new parameters to `tweakreg_step` star object detection algorithm to allow source selection/filtering by:

1. `peakmax` - maximum peak pixel value for selecting stars.
2. `brightest` - number of brightest objects to keep.

The rationale for these parameters is that `peakmax` could allow automatic exclusion of sources that reach a known saturation level of the detector and `brightest` would keep top `brightest` stars thus eliminating the faintest objects that could be false detections.

CC: @hbushouse @larrybradley

hbushouse commented on jwst/tweakreg/tweakreg_catalog.py:
```diff
@@ -4,7 +4,8 @@
 def make_tweakreg_catalog(model, kernel_fwhm, snr_threshold, sharplo=0.2,
-                          sharphi=1.0, roundlo=-1.0, roundhi=1.0):
+                          sharphi=1.0, roundlo=-1.0, roundhi=1.0,
+                          brightest=None):
```
Need to add peakmax to the argument list here, so that it's defined for use in line 72.

mcara replied:
Thank you!

hbushouse commented on jwst/tweakreg/tweakreg_catalog.py:
```diff
@@ -52,7 +68,8 @@ def make_tweakreg_catalog(model, kernel_fwhm, snr_threshold, sharplo=0.2,
                             sharplo=sharplo, sharphi=sharphi, roundlo=roundlo,
-                            roundhi=roundhi)
+                            roundhi=roundhi, brightest=brightest,
+                            peakmax=peakmax)
```
The value "peakmax" is currently undefined at this point in the code

hbushouse commented on jwst/tweakreg/tweakreg_step.py:
```diff
@@ -27,9 +27,11 @@ class TweakRegStep(Step):
+        kernel_fwhm = float(default=2.5) # Gaussian kernel FWHM in pixels
+        snr_threshold = float(default=10.0) # SNR threshold above the bkg
+        brightest = int(default=None) # Keep top ``brightest`` objects
+        peakmax = float(default=None) # Filter out objects with pixel values >= ``pixmax``
```
"pixmax" should be "peakmax"

hbushouse commented on jwst/tweakreg/tweakreg_step.py:
```diff
@@ -27,9 +27,11 @@ class TweakRegStep(Step):
+        kernel_fwhm = float(default=2.5) # Gaussian kernel FWHM in pixels
+        snr_threshold = float(default=10.0) # SNR threshold above the bkg
+        brightest = int(default=None) # Keep top ``brightest`` objects
+        peakmax = float(default=None) # Filter out objects with pixel values >= ``peakmax``
```
The /pipeline/tweakreg.cfg file should also be updated to include these parameter changes (and to make sure they're consistent with one another).

mcara commented:
@hbushouse I think I have addressed the issues pointed out by you. Thanks!

I also want to ask if you would like to limit number of objects used for alignment by setting the default value of `brightest` to something like `50`, or `100`, etc. Or, will this be handled differently in the pipeline in an instrument-specific manner? I think even 50 objects should be enough for good image alignment (unless all these sources are clustered in a single corner of the image). All of this is to make object detection less susceptible to "fake objects" (i.e., hot pixels, random structure in the noise, etc.)

mcara commented:
@hbushouse I keep forgetting about `tweakreg.cfg`...

mcara commented:
@hbushouse Also, IMO, a minimum of 3 sources are necessary for image alignment using 'general' (shifts, rotations, scale, skew) fit. Should I lower `minobj` to 3? In fact we can lower it to 1 because I think the algorithm will automatically downgrade `fitgeom` to the appropriate level (i.e., from 'general' to 'rscale' to 'shifts').

mcara commented:
@nden Done with @larrybradley help

mcara commented:
@hbushouse  astropy/photutils#750
https://github.com/astropy/photutils/pull/750 was merged

hbushouse commented:
@stscieisenhamer @mcara Any ideas as to why lib/tests/test_suffix.py is throwing an error in the Travis tests? Seems to be a mismatch in "tweakreg" suffix. How would these updates be causing that?

stscieisenhamer commented:
Not exactly sure how the current changes affect this. I'm guessing there is a change upstream in the 0.13.0_new_features that actually did this.

Regardless, all that is needed is to remove `tweakreg` from the list of calculated suffixes in `lib/suffix.py`.
